### ISSUE TITLE
Name of the place update

### ISSUE THREAD

**Issue by questionablequestion** [whosonfirst-data/whosonfirst-data#1997]
Hello,

I'm wondering if I you can let me know where is data for Bosnia and Herzegovina fetched from? And how is it possible to rename some names as they are not really official names (some county names). Is editing the .geojson and making PR sufficient enough or how is that handled? I can't really figure out based on the readme.

Thanks

**Comment by nvkelso** [comment]
@stepps00 can you give @questionablequestion a few tips, please?

**Comment by stepps00** [comment]
Hey @questionablequestion you can poke around Who's On First records using the [Spelunker tool](https://spelunker.whosonfirst.org/search/?q=Bosnia+and+Herzegovina). You'll find source information for geometries in individual records' `src:*` properties. Check out [this record](https://spelunker.whosonfirst.org/id/85669235/) as an example.

Names come from a variety of sources, namely Wikidata, Natural Earth, Quattroshapes, and GeoNames.. though there are plenty of others. To see a full list of sources used in Who's On First, take a look here: https://github.com/whosonfirst/whosonfirst-sources/tree/main/sources#sources

If you have suggestions for name updates, feel free to submit a PR. Alternatively, if you have a list of Who's On First IDs (`wof:id` property) and updated name pairs, we can update a set of records in a PR and flag you for review.

**Comment by thisisaaronland** [comment]
@questionablequestion Records in Who's On First are never deleted. Rather if a record is considered invalid or incorrect (or otherwise "bunk") it is marked as deprecated.

Specifically it is assigned a `edtf:deprecated=YYYY-MM-DD` property. If a deprecated record has been replaced by a newer record then it is considered "superseded" and is assigned a `wof:superseded_by=[ NEW_ID ]` property. The newer record is also assigned a `wof:supersedes=[ OLD_ID ]` property.

Related: https://whosonfirst.org/blog/2016/10/06/wof-lifecycle-documentation/

**Comment by questionablequestion** [comment]
Good to know, thank you @thisisaaronland

**Comment by questionablequestion** [comment]
How to properly add a new entries (like a new locality)?

Also I find something confusing, if you can help me:
[openstreetmap:venue:way/475677703](https://www.openstreetmap.org/way/475677703) is showing that the county is [1108759181](https://spelunker.whosonfirst.org/id/1108759181/) but it should be [1108759047](https://spelunker.whosonfirst.org/id/1108759047/). And that same problem is repeating for some more locations in different counties.

I'm using pelias and on their [demo](https://pelias.github.io/compare/#/v1/autocomplete?lang=bs&text=skenderija+most), it's the same problem. Is this caused by pelias somehow or is this issue with WOF?

**Comment by questionablequestion** [comment]
@stepps00  @thisisaaronland Can you please help me with how to make proper changes? I have been checking out other PR's and they have dates that seems like they are generated (with differences in ms in between them), are they actually or do I just get current timestamp and paste it there? Also roughly how long does it take for changes to be released?

I can if you still want me to make a list with changes and new records to be added if you think that would be easier and faster.

Thanks!

**Comment by stepps00** [comment]
> @stepps00 Thanks! So Changes are done manually by modifying .geojson files in the country specific repo? Most of changes I would need is just change of the name_loc and name_en property.

Yes - changes to records would be made to the geojson files in the Who's On First admin repo(s), which are split by iso-country code. The Norway admin repo, as an example:

https://github.com/whosonfirst-data/whosonfirst-data-admin-no

> Also I find something confusing, if you can help me: openstreetmap:venue:way/475677703 is showing that the county is 1108759181 but it should be 1108759047. And that same problem is repeating for some more locations in different counties. I'm using pelias and on their demo, it's the same problem. Is this caused by pelias somehow or is this issue with WOF?

If I understand this correctly, the location of this bridge you're linking to in Sarajevo falls within the geometry of 1108759181, rather than 1108759047 when using the Pelias service. This is likely because the precision of the  geometries in Who's On First are slightly off from official boundaries. 

> @stepps00 @thisisaaronland Can you please help me with how to make proper changes? I have been checking out other PR's and they have dates that seems like they are generated (with differences in ms in between them), are they actually or do I just get current timestamp and paste it there?

When updating records, Who's On First requires openly licensed data (under a `CC-BY` or `CC0`- equivalent license). Are you thinking of updates to a specific area / country, like Bosnia and Herzegovina? There are a few steps to updating records, which include:

- Cataloguing the data source in the [whosonfirst-sources](https://github.com/whosonfirst/whosonfirst-sources) repo
- Storing [alt-categories](https://github.com/whosonfirst/whosonfirst-cookbook/blob/main/how_to/creating_alt_geometries.md) appropriately
- Updating properties as needed
- Updating hierarchies as needed (info [here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/main/how_to/setting_up_pip_service.md))

That being said, if you have a data source that can be used to update Who's On First records, we can chat about specifics ad help craft a PR.

> Also roughly how long does it take for changes to be released?

Once changes are merged to Who's On First in GitHub, they are available to use immediately. However, services like Pelias will pull those changes in at various intervals. Who's On First distributions hosted by Geocode Earth are here, and built ~daily: https://geocode.earth/data/whosonfirst/

**Comment by questionablequestion** [comment]
@stepps00 Thanks! I have already made changes to geojson files (Bosnia and Herzegovina) and I will make a PR on those first before attempting at creating new records (they are mostly neighbourhoods and some changing from locality to neighbourhood). I still would have to figure out if there's any source to take boundaries (geometry) for those new neighbourhoods.

About that misalignment, I'm still not sure how does it falls within wrong geometry when it's not even in the boundary of that wrong county? Is it there anything I can do? Sarajevo is pretty small city (compared to other European cities) so mistakes like this can give wrong result when searching for something that is in that area and that's an issue for me.

**Comment by stepps00** [comment]
Sounds good, feel free to flag me in any PRs, I can review the change(s).

> About that misalignment, I'm still not sure how does it falls within wrong geometry when it's not even in the boundary of that wrong county?

How are you determining "openstreetmap:venue:way/475677703 is showing that the county is 1108759181 but it should be 1108759047."? Is this an issue you're seeing with Pelias? I'm assuming this is due to Who's On First geometries because these records have simplified geometries.. but there may be additional data or parameters in Pelias that are not maintained in Who's On First.

**Comment by questionablequestion** [comment]
@stepps00 Yes, I see that issue in Pelias and I'm guessing I should ask them instead as they are probably calculating that in the background.

Another question I have is, Bosnia and Herzegovina is contained with two entities (Federation of BiH and Republic of Srpska) and Brcko Distrikt. Currently, there's no Federation of BiH record in Who's On First. There is deprecated record (890519107) but no substitution for that. I think it would be a good idea to add Federation record as well and I was planning on adding neighbourhoods where I could also add Federation and link everything up. But my question is what's the way with adding new records? I was planning on using OSM as there's data/geometries I need already there. Can you please send me a link for that guide or should I write some script to scrape specific OSM ids and create geojson files in WOF? Thanks!

**Comment by thisisaaronland** [comment]
@questionablequestion WOF does not import any OSM data due to the former's licensing restrictions. In software licensing terms OSM is akin to the GPL and WOF is like the MIT or BSD licenses.

**Comment by questionablequestion** [comment]
@thisisaaronland Oh, okay, I saw in the sources that there's osm source defined and I guessed it was being used. So what would my options be for the geometry if there's no other source I could use? Would tracing boundaries by myself be good enough to be added in WOF? Neighbourhoods are in questions so they are not big nor complex. That would mean much more labor work but I would do that if there's not other way.

**Comment by stepps00** [comment]
@questionablequestion if we're able to find an openly licensed dataset to import, that would be great.. but if not, Who's On First can use hand-curated geometries to update place records. Ideally the geometries would have a `wof:id` attribute associated with them to simplify the import.
